request_helpers.go

```diff
@@ -40,15 +37,20 @@ func getPaginatedResults[T any](
 		opts.PageOptions = &PageOptions{Page: 0}
 	}
 
-	// Apply all user-provided list options to the base request
-	if err := applyListOptionsToRequest(opts, req); err != nil {
-		return nil, err
-	}
-
 	// Makes a request to a particular page and
 	// appends the response to the result
 	handlePage := func(page int) error {
-		req.SetQueryParam("page", strconv.Itoa(page))
+		// Override the page to be applied in applyListOptionsToRequest(...)
+		opts.Page = page
+
+		// This request object cannot be reused for each page request
+		// because it can lead to possible data corruption
+		req := client.R(ctx).SetResult(resultType)
```

Comment:
Another potential solution could've been to call `req.SetResult(...)` at the beginning of this function but I wasn't 100% sure that wouldn't have unintended side-effects